langlib/lang.array/src/main/ballerina/array.bal

```diff
@@ -130,6 +130,21 @@ public isolated function slice(Type[] arr, int startIndex, int endIndex = arr.le
     name: "slice"
 } external;
 
+# Tests whether a function returns true for every member of an array.
+# `func` is called for each member of `arr` in order unless and until a call returns false.
+# + arr - the array
+# + func - function to apply to each member
+# + return - true if func returns true for every member of `arr`; otherwise, false
+#
+public isolated function every(Type[] arr, @isolatedParam function(Type val) returns boolean func) returns boolean {
+    foreach var item in arr {
+        if !func(item) {
+            return false;
+        }
+    }
+    return arr.length() != 0;
```

Comment:
Are we returning false when the list is empty because the docs for the returned value says
```
true if func returns true for every member of `arr`; otherwise, false
```

Should we get this clarified in the spec? Java, JavaScript, etc. return true IINM. My initial thoughts looking at this were also that the test is on members, if any. So if there are no members, it will be true.

cc @hasithaa

Reply:
It is verified in https://github.com/ballerina-platform/ballerina-spec/issues/1005#issuecomment-1076020981